dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java

```diff
@@ -0,0 +1,113 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.dropbox;
+
+import java.io.InputStream;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.dropbox.core.v2.DbxClientV2;
+import org.apache.commons.io.IOUtils;
+import org.jbpm.document.Document;
+import org.jbpm.document.service.impl.DocumentImpl;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "DropboxDownloadFileDefinitions.wid", name = "DropboxDownloadFile",
+        displayName = "DropboxDownloadFile",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.dropbox.DownloadFileWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "Path"),
+                @WidParameter(name = "DocumentName")
+        },
+        results = {
+                @WidResult(name = "Document")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "com.dropbox.core", artifact = "dropbox-core-sdk", version = "3.0.5")
+        })
+public class DownloadFileWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(DownloadFileWorkitemHandler.class);
+    private static final String RESULTS_DOCUMENT = "Document";
+
+    private DropboxAuth auth;
+    private DbxClientV2 client;
+    private String clientIdentifier;
+    private String accessToken;
+
+    public DownloadFileWorkitemHandler(String clientIdentifier,
+                                       String accessToken) {
+        this.clientIdentifier = clientIdentifier;
+        this.accessToken = accessToken;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        Map<String, Object> results = new HashMap<String, Object>();
+
+        try {
+
+            if (auth == null) {
+                auth = new DropboxAuth();
+            }
+
+            client = auth.authorize(clientIdentifier,
+                                    accessToken);
+
+            String dropboxPath = (String) workItem.getParameter("Path");
+            String documentName = (String) workItem.getParameter("DocumentName");
```

Comment:
wouldn't it be better to have single param "path" that points at file to download?